libexfat/io.c

```diff
@@ -76,6 +87,34 @@ static int open_rw(const char* spec)
 	return fd;
 }
 
+__attribute__((format(scanf, 4, 5)))
+static int read_sys_block_attr(int dev_major, int dev_minor, const char* attr,
+		const char* format, ...)
+{
+	char path[128];
+	FILE *fp;
+	va_list args;
+	int rc;
+
+	snprintf(path, sizeof(path), "/sys/dev/block/%d:%d/%s", dev_major, dev_minor, attr);
+	fp = fopen(path, "r");
+	if (!fp)
+	{
+		snprintf(path, sizeof(path), "/sys/dev/block/%d:%d/../%s", dev_major, dev_minor, attr);
```

Comment:
Why does this path contain `..`?

Reply:
Because it's possible the attributes might not be present on a device representing a partition but do exist on its parent disk.